be/src/exprs/cast_expr.cpp

```diff
@@ -237,6 +251,42 @@ static ColumnPtr cast_from_json_fn(ColumnPtr& column) {
     return builder.build(column->is_constant());
 }
 
+template <LogicalType FromType, LogicalType ToType, bool AllowThrowException>
+static ColumnPtr cast_from_variant_fn(ColumnPtr& column) {
+    ColumnViewer<TYPE_VARIANT> viewer(column);
```

Comment:
Is `FromType` always `TYPE_VARIANT` here?

Reply:
Yes. Only invoke this method when `FromType` is `VARIANT`